pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -168,7 +174,7 @@ func toEnvoyEndpoint(e *model.IstioEndpoint, push *model.PushContext) (*endpoint
 	// Istio telemetry depends on the metadata value being set for endpoints in the mesh.
 	// Istio endpoint level tls transport socket configuation depends on this logic
 	// Do not remove
-	ep.Metadata = util.BuildLbEndpointMetadata(e.UID, e.Network, e.TLSMode, push)
+	ep.Metadata = util.BuildLbEndpointMetadata(e.UID, e.Network, e.EffectiveTlsMode, push)
```

Comment:
I don't get why you need to use EffectiveTlsMode here. Can't we just compute that here?

Reply:
see the other comment that i answered @hzxuzhonghu